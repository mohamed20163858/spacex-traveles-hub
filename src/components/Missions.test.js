import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import '@testing-library/jest-dom';
import MissionTableRow from './missionTableRow';

function TestMissions() {
  const id = 'EE86F74';
  const mssionname = 'Commercial Resupply Services';
  const description = 'discription';
  const reserved = true;
  const key = 'mission-4';
  return (
    <Provider store={store}>
      <MissionTableRow
        missionId={id}
        missionName={mssionname}
        description={description}
        reserved={reserved}
        key={key}
      />
    </Provider>
  );
}
it('renders correctly', () => {
  const tree = renderer
    .create(<TestMissions />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
test('simulate checking a joined mission ', () => {
  // ARRANGE
  render(<TestMissions />);

  // ACT making reserve = true

  // Arrange
  const result = screen.getByText('Active Member');
  // ASSERT
  expect(result).toBeInTheDocument();
});
