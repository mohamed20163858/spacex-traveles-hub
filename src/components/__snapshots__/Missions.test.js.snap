// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<tr
  id="EE86F74"
>
  <td>
    Commercial Resupply Services
  </td>
  <td>
    <p>
      discription
    </p>
  </td>
  <td>
    <p
      className="badge-2"
    >
      Active Member
    </p>
  </td>
  <td>
    <button
      className="reserved"
      onClick={[Function]}
      type="submit"
    >
      Leave Mission
    </button>
  </td>
</tr>
`;
